File: myapplication/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import com.example.reliablemessaging.ReliableMessaging;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Map<String, String> data = new HashMap<>();
        findViewById(R.id.btnSend).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String text = ((EditText) findViewById(R.id.message)).getText().toString();
                if (text.isEmpty()) return;
                data.put("https://challenge.ronash.co/reliable-messaging", text);
                ReliableMessaging.sendMessage(data);
            }
        });

    }
}

File: app/src/main/java/david/hosseini/android_reliable_messaging_challenge/MainActivity.kt
package david.hosseini.android_reliable_messaging_challenge

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.work.WorkInfo
import david.hosseini.androidreliablemessaginglibrary.post
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun send(view: View) {

        val text = edit_text.text.toString()
        post(URL, mapOf(TEXT to text))
                .observe(this, Observer<WorkInfo> {

                    if (it.state == WorkInfo.State.SUCCEEDED) {

                        Toast.makeText(
                                this, "Request successfully finished",
                                Toast.LENGTH_SHORT
                        ).show()
                    }
                })

    }
}

const val URL = "https://challenge.ronash.co/reliable-messaging/"
const val TEXT = "text"